### PR TITLE
Fix warnings in plugin tests by checking if an ivar was defined

### PR DESCRIPTION
**Pull request by ayushn21**
This is a 🐛 bug fix. 

## Summary

One of my plugins' tests were logging the following lines:

`frontmatter_defaults.rb:16: warning: instance variable @glob_cache not initialized`
`localizable.rb:8: warning: instance variable @locale not initialized`

This PR fixes those by checking if those instance variables were `defined?` rather than if they exist.

### CODE REVIEW

**Comment by ayushn21** [comment]
I'm not sure what's up with that test failure @jaredcwhite. I can't reproduce it locally and I've rerun the tests a couple of times already. Any ideas?

**Comment by jaredcwhite** [comment]
Hmm, sounds like somehow Ruby's warning level was set to the most strict. I normally don't see any warnings about missing ivars in my Ruby projects.

**Comment by ayushn21** [comment]
It was probably set to the default for new Bridgetown plugins. I'm not sure I even know how to change the warning level :P 

I reckon it's worth fixing though because checking an instance variable before it's been defined seems a bit iffy in my mind 🤷 

Not too fussed either way though.

File: bridgetown-core/lib/bridgetown-core/concerns/site/localizable.rb [hunk 1]
```diff
@@ -5,7 +5,7 @@ module Localizable
     # Returns the current and/or default configured locale
     # @return String
     def locale
-      if @locale
+      if defined?(@locale)
```
Comment (jaredcwhite) [inline bridgetown-core/lib/bridgetown-core/concerns/site/localizable.rb:8]: let's just memoize this via a `@locale ||= begin` block

File: bridgetown-core/lib/bridgetown-core/frontmatter_defaults.rb [hunk 2]
```diff
@@ -13,7 +13,7 @@ def initialize(site)
     end
 
     def reset
-      @glob_cache = {} if @glob_cache
+      @glob_cache = {} if defined?(@glob_cache)
```
Comment (jaredcwhite) [inline bridgetown-core/lib/bridgetown-core/frontmatter_defaults.rb:16]: With a second look at this, I don't think it needs the `if` statement at all actually.

**Review by jaredcwhite — CHANGES_REQUESTED** [review]
Couple suggestions on alternate fixes…

File: bridgetown-core/lib/bridgetown-core/concerns/site/localizable.rb (hunk 1 above)
Comment (ayushn21) [reply bridgetown-core/lib/bridgetown-core/concerns/site/localizable.rb]: Ah nice, I didn't know you could use `begin` / `end` in this way!

**Comment by ayushn21** [comment]
Updated the PR based on your feedback @jaredcwhite :)

**Pull request merged**